### PR TITLE
Added two workflows and .editorconfig

### PR DESCRIPTION
**Pull request by SeniorMars**
The workflows are:
    - dependabot
    - check

Dependabot just checks that dependencies are safe and upgraded if needed.

Check adds a bunch of things I feel are needed.

Note: Both of these workflows come from: https://github.com/jonhoo/rust-ci-conf

An .EditorConfig is provide as it is very important to have the same style everywhere regardless of editor.

### CODE REVIEW

**Comment by SeniorMars** [comment]
Note: check clippy fails right here because this pr is not yet merged as such doesn't have permissions.

**Comment by SeniorMars** [comment]
Also, you should probably turn off allowing workflows to run if there are from public forks: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

**Comment by laurmaedje** [comment]
I like how light the CI is currently. Personally, I also don't really like dependabot and clippy. The two things I would be fine with is checking formatting and docs. I think these could be integrated into the existing `ci.yml` workflow.

**Comment by SeniorMars** [comment]
I deleted the clippy and hack job and added the rest to `ci.yml`. I kept `msrv` (minimal supported version) as I feel it is useful. Please let me know if there is anything else I can change.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,3 +15,47 @@ jobs:
...
+    name: stable / fmt
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Install stable
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt
+      - name: cargo fmt --check
```
Comment (drupol) [inline .github/workflows/ci.yml:29]: How about using a name instead of using the command here?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,3 +15,47 @@ jobs:
...
+  doc:
+    runs-on: ubuntu-latest
+    name: nightly / doc
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Install nightly
+        uses: dtolnay/rust-toolchain@nightly
+      - name: cargo doc
```
Comment (drupol) [inline .github/workflows/ci.yml:40]: `Build doc` instead ?

**Review by drupol — COMMENTED** [review]
Added minor syntactic comments

**Comment by laurmaedje** [comment]
I added a formatting check to CI and an editor configuration.

**Pull request closed without merging**